### PR TITLE
Switch to inversify, add docs...

### PR DESCRIPTION
**Pull request by joelday**
chore: enable strict mode in tsconfig.json and fix resulting compile-time null/undefined and typing errors.
chore: Improvements to eventToValueObservable handling and typing of undefined.
chore: Removing ancient legacy debug plugin upgrade check.
fix: Ensure that language service status items are updated properly.
feat: Updated welcome message with debugger upgrade info.
fix: Various potential extension errors/crashes in rare circumstances. (Empty config values, etc.)
feat: Update display name for Skyrim to include AE.
chore: Replace decoration-ioc with inversify (actually maintained and not broken)
docs: Add notes about updated debugger, Address Library requirements.
chore: Add CI build version to debugger dlls.
chore: Use Release build configuration for CI builds.

### CODE REVIEW

**Comment by joelday** [comment]
I need to add the welcome message logic to open based on a comparison with the previously installed version.

File: src/papyrus-lang-vscode/src/server/LanguageClientHost.ts [hunk 1]
```diff
@@ -65,10 +65,11 @@ export class LanguageClientHost implements ILanguageClientHost, Disposable {
         this._projectInfos = this._status.pipe(
             switchMap((status) => {
                 if (status !== ClientHostStatus.running) {
-                    return Promise.resolve(null);
+                    return of(null);
```
Comment (nikitalita) [inline src/papyrus-lang-vscode/src/server/LanguageClientHost.ts:68]: of….?

Comment (nikitalita) [reply src/papyrus-lang-vscode/src/server/LanguageClientHost.ts:68]: Oh, it’s reactive stuff.

File: src/papyrus-lang-vscode/package.json [hunk 2]
```diff
@@ -99,9 +99,9 @@
                         }
                     },
                     {
-                        "label": "Papyrus: Skyrim Special Edition (with .ppj)",
+                        "label": "Papyrus: Skyrim Special Editio/Anniversary Editionn (with .ppj)",
```
Comment (nikitalita) [inline src/papyrus-lang-vscode/package.json:102]:
```suggestion
                        "label": "Papyrus: Skyrim Special Edition/Anniversary Edition (with .ppj)",
```

**Review by nikitalita — CHANGES_REQUESTED** [review]

**Review by nikitalita — APPROVED** [review]

**Pull request merged**